lib/charcoal/jsonp.rb

```diff
@@ -38,8 +42,8 @@ def add_jsonp_callback
       yield
 
       if response.status.to_s.starts_with?('200') && jsonp_request?
-        response.body = "#{params[:callback]}(#{response.body})"
         response.content_type = "application/javascript"
+        response.body = "#{params[:callback]}(#{response.body.to_s})"
```

Comment:
I don't think this works with streamed responses anyway. Is this something to bring up as an issue upstream?